lib/common/logging.c

```diff
@@ -135,11 +135,17 @@ crm_log_deinit(void)
  * \internal
  * \brief Set the log format string based on the passed-in method
  *
- * \param[in] method  The detail level of the log output
- * \param[in] daemon  The daemon ID included in error messages
+ * \param[in] method        The detail level of the log output
+ * \param[in] daemon        The daemon ID included in error messages
+ * \param[in] use_pid       Cached result of getpid() call, for efficiency
+ * \param[in] use_nodename  Cached result of uname() call, for efficiency
+ *
  */
+
+/* XXX __attribute__((nonnull)) for use_nodename parameter */
```

Comment:
I think I missed a discussion the other day in our meeting about using the nonnull attribute.  Personally, I am a fan of using attributes wherever we can to get the compiler to do more work for us.  If the concern is that not all compilers support this attribute, we can add a check in configure.ac to test if it's supported, and then define something in a header file based on that.  Something like:

```
#if HAS_ATTR_NONNULL
#define ATTR_NONNULL(...) __attribute__((__VA_ARGS__))
#else
#define ATTR_NONNULL
#endif
```

Or something like that.  vararg macros are weird.  Anyway, then you would just use ATTR_NONNULL(...) instead.